overlord/snapstate/snapmgr.go

```diff
@@ -309,38 +352,39 @@ func (m *SnapManager) doDownloadSnap(t *state.Task, _ *tomb.Tomb) error {
 
 	meter := &TaskProgressAdapter{task: t}
 
-	var auther store.Authenticator
-	if ss.UserID > 0 {
-		st.Lock()
-		user, err := auth.User(st, ss.UserID)
-		st.Unlock()
-		if err != nil {
-			return err
-		}
-		auther = user.Authenticator()
-	}
-
-	storeInfo, err := m.store.Snap(ss.Name, ss.Channel, ss.DevMode(), auther)
+	st.Lock()
+	store := Store(st)
+	auther, err := autherForUserID(st, ss.UserID)
+	st.Unlock()
 	if err != nil {
 		return err
 	}
 
-	if err = checkRevisionIsNew(ss.Name, snapst, storeInfo.Revision); err != nil {
-		return err
+	// compatiblity with old tasks
```

Comment:
typo, also I would move this comment inside if branch or closer to the if line at least?